TCAT/Models/WalkPath.swift

```diff
@@ -28,10 +28,38 @@ class WalkPath: Path {
         self.path = untraveledPath
         self.strokeColor = color
         self.strokeWidth = polylineWidth
+        
+        guard let path = self.path else { return }
+        let intervalDistanceIncrement: CGFloat = 20
+        var previousCircle: (coordinate: CLLocationCoordinate2D, radius: Double)?
+        // Maps circle coordinates in incremental distance
+        for coordinateIndex in 0 ..< path.count() - 1 {
+            let startCoordinate = path.coordinate(at: coordinateIndex)
+            let endCoordinate = path.coordinate(at: coordinateIndex + 1)
+            let startLocation = CLLocation(latitude: startCoordinate.latitude, longitude: startCoordinate.longitude)
+            let endLocation = CLLocation(latitude: endCoordinate.latitude, longitude: endCoordinate.longitude)
+            let pathDistance = endLocation.distance(from: startLocation)
+            let intervalLatIncrement = (endLocation.coordinate.latitude - startLocation.coordinate.latitude) / pathDistance
+            let intervalLngIncrement = (endLocation.coordinate.longitude - startLocation.coordinate.longitude) / pathDistance
+
+            for intervalDistance in 0 ..< Int(pathDistance) {
+                let intervalLat = startLocation.coordinate.latitude + (intervalLatIncrement * Double(intervalDistance))
+                let intervalLng = startLocation.coordinate.longitude + (intervalLngIncrement * Double(intervalDistance))
+                let circleCoordinate = CLLocationCoordinate2D(latitude: intervalLat, longitude: intervalLng)
 
-        self.spans = GMSStyleSpans(untraveledPath!, [.solidColor(self.color)], dashLengths, .projected)
-        self.geodesic = false
+                if let previousCircle = previousCircle {
+                    let circleLocation = CLLocation(latitude: circleCoordinate.latitude, longitude: circleCoordinate.longitude)
+                    let previousCircleLocation = CLLocation(latitude: previousCircle.coordinate.latitude, longitude: previousCircle.coordinate.longitude)
 
+                    if circleLocation.distance(from: previousCircleLocation) < intervalDistanceIncrement {
+                        continue
+                    }
```

Comment:
If your condition is very small (usually 1-2 words / keywords) make it a single line:

`if circleLocation.distance(from: previousCircleLocation) < intervalDistanceIncrement { continue }`